### ISSUE TITLE
Add-Migration mysteriously retrieves old migration information

### ISSUE THREAD

**Issue by sjb-sjb** [dotnet/efcore#28316]
I changed my model, and I wanted to start my migrations over again from scratch instead of adding a new migration. 
I deleted the existing migration file and the corresponding .Designer.cs file. 
I am using Sqlite and choose the file location dynamically, so there is no database copy maintained and no history migrations table to delete. I did delete the old database anyway.

I ran Add-Migration and it somehow created a .Designer.cs file that referred to the old model, and the new migration file was the difference from that to the current model. 

I am at a loss for how EF could even do that given I deleted the old .Designer and migration files. Doing a Clean Solution, restarting VS, and checking in the project after deleting the old migration did not make any difference. 

How can I clear out whatever cache it is that Add-Migration is using?

**Comment by roji** [comment]
@sjb-sjb EF Core doesn't have any particular cache - if you delete your Migrations folders in your source directory, the next Add-Migration should create a new, initial migration for your model, exactly as it is at that point.

If you're seeing another behavior, please submit a minimal project with instructions on how to repro the problematic state.

**Comment by sjb-sjb** [comment]
Yeah that's what I thought but the observed behavior was as if the Migrations directory did not contain all the cached model information. And what I found out was that in fact it doesn't! So it's a bug. 

Steps to reproduce: Use both -Namespace and -OutputDir options on the Add-Migrations command. The *ModelSnapshot.cs file is produced in a directory based on the Namespace while the migration files are put into the OutputDir. They should both go into the migrations OutputDir.

Not sure what the *ModelSnapshot.cs is needed for given that it is almost identical to the *.Designer.cs file that is in Migrations directory. Is there a way to suppress it?

**Comment by sjb-sjb** [comment]
@ajcvickers thanks for the note. For that matter thanks to the whole team for all the good work on EF.

**Comment by Shakerrida** [comment]
I don't know if this is another bug or not, but if the namespace contains a dot it will be replaced with a path separator for the snapshot file only.

here is a sample json:
{
  "migrationFile": "C:\\Work\\Git\\Migration\\UserMigrations\\Migrations\\20230517070603_test.cs",
  "metadataFile": "C:\\Work\\Git\\Migration\\UserMigrations\\Migrations\\20230517070603_test.Designer.cs",
  "snapshotFile": "C:\\Work\\Git\\Migration\\Maestro\\UserMigrations\\DynamicDbContextPostgresModelSnapshot.cs"
}

**Comment by bricelam** [comment]
@Shakerrida That looks like a different bug. Could you file a new issue? Please include a simple project we can use to reproduce the issue and the exact command you run.

**Comment by ajcvickers** [comment]
Note from design meeting: if you specify output-path, then we will respect it. See also https://github.com/dotnet/efcore/issues/32738.

**Comment by sjb-sjb** [comment]
Great, glad this will get fixed!

**Comment by coderbarncb** [comment]
I wanted to add that the bug that @Shakerrida reported still exists.
Example:

`dotnet ef migrations add InitialCreate --namespace 'my.project.Infra.Db.Migrations' --project 'my.project.Infra' --startup-project 'my.project.ApiSvc' --context 'my.project.Infra.Db.MyDbContext' -o 'Db/Migrations'`

This will correctly put the migrations / designer file(s) into `my.project.Infra/Db/Migrations`, however, the
`MyDbContextModelSnapshot.cs` file will end up in this directory: `my.project.Infra/my/project/Db/Migrations`
